appserver/packager/pom.xml

```diff
@@ -298,80 +471,51 @@
-                                        <mapper type="glob" from="*.pack.gz" to="*.jar" />
-                                    </apply>
-                                    <!--remove *.pack.gz files-->
-                                    <delete>
-                                        <fileset dir="${stage.dir}/glassfish/modules" includes="**/*.pack.gz" />
-                                    </delete>
-                                </target>
-                            </configuration>                         
-                        </execution>
                     </executions>
                 </plugin>
+                
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
-                    <artifactId>maven-resources-plugin</artifactId>
+                    <artifactId>maven-javadoc-plugin</artifactId>
                     <configuration>
-                        <includeEmptyDirs>true</includeEmptyDirs>                    
-                    </configuration>                    
-                    <executions>
-                        <execution>
-                            <id>copy-resources</id>
-                            <phase>validate</phase>
-                            <goals>
-                                <goal>copy-resources</goal>
-                            </goals>
-                            <configuration>
-                                <outputDirectory>${basedir}/..</outputDirectory>
-                                <overwrite>true</overwrite>
-                                <resources>          
-                                    <resource>
-                                        <directory>${basedir}/../resources</directory>
-                                        <filtering>true</filtering>
-                                        <includes>
-                                            <include>pkg_conf.py</include>
-                                        </includes>
-                                    </resource>
-                                </resources>              
-                            </configuration>            
-                        </execution>
-                    </executions>
+                        <skip>true</skip>
+                    </configuration>
                 </plugin>
             </plugins>
         </pluginManagement>
     </build>
-</project>
+    
+    <!-- Alternative profile: embed the sources into each dependency (for trivial source lookup in IDEs) -->
+    <profiles>
+        <profile>
+            <id>embed-source</id>
+            <activation><activeByDefault>false</activeByDefault></activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-dependency-plugin</artifactId>
+                        <executions>
```

Comment:
Given the packaging process is complicated already, wouldn't be better to define those source-only steps in profile only?

Reply:
Lines 495 to 498 are in the embed-source profile, so the source-only steps are already in profile. Or did you mean something else?

Reply:
Yes, whether you considered configuration of this execution to be put here as well. The `<phase>none</phase>` is easy to miss when reasoning about what the actual process is.

Reply:
I put it there since that's how the IPS profile works too. All the config in one place, and then the profiles activate it.